src/simplewallet/simplewallet.cpp

```diff
@@ -3742,101 +3782,120 @@ bool simple_wallet::init(const boost::program_options::variables_map& vm)
+          else
+            heightstr = input_line("Restore from specific blockchain height (optional, default 0),\nor alternatively from specific date (YYYY-MM-DD)");
+          if (std::cin.eof())
+            return false;
+          if (heightstr.empty())
           {
-            fail_msg_writer() << tr("bad m_restore_height parameter: ") << heightstr;
-            continue;
+            m_restore_height = 0;
+            break;
           }
-          uint16_t year;
-          uint8_t month;  // 1, 2, ..., 12
-          uint8_t day;    // 1, 2, ..., 31
           try
           {
-            if (!datestr_to_int(heightstr, year, month, day))
-              return false;
-            m_restore_height = m_wallet->get_blockchain_height_by_date(year, month, day);
-            success_msg_writer() << tr("Restore height is: ") << m_restore_height;
-            std::string confirm = input_line(tr("Is this okay?"), true);
-            if (std::cin.eof())
-              return false;
-            if(command_line::is_yes(confirm))
-              break;
+            m_restore_height = boost::lexical_cast<uint64_t>(heightstr);
+            break;
           }
           catch (const boost::bad_lexical_cast &)
           {
-            fail_msg_writer() << tr("bad m_restore_height parameter: ") << heightstr;
-          }
-          catch (const std::runtime_error& e)
-          {
-            fail_msg_writer() << e.what();
+            if (!connected || version < MAKE_CORE_RPC_VERSION(1, 6))
+            {
+              fail_msg_writer() << tr("bad m_restore_height parameter: ") << heightstr;
+              continue;
+            }
+            uint16_t year;
+            uint8_t month;  // 1, 2, ..., 12
+            uint8_t day;    // 1, 2, ..., 31
+            try
+            {
+              if (!datestr_to_int(heightstr, year, month, day))
+                return false;
+              m_restore_height = m_wallet->get_blockchain_height_by_date(year, month, day);
+              success_msg_writer() << tr("Restore height is: ") << m_restore_height;
+              std::string confirm = input_line(tr("Is this okay?"), true);
+              if (std::cin.eof())
+                return false;
+              if(command_line::is_yes(confirm))
+                break;
+            }
+            catch (const boost::bad_lexical_cast &)
+            {
+              fail_msg_writer() << tr("bad m_restore_height parameter: ") << heightstr;
+            }
+            catch (const std::runtime_error& e)
+            {
+              fail_msg_writer() << e.what();
+            }
           }
         }
       }
     }
+    
     if (m_restoring)
     {
-      uint64_t estimate_height = m_wallet->estimate_blockchain_height();
-      if (m_restore_height >= estimate_height)
+      if (seed.size() == sizeof(crypto::secret_key&) * 26 / 32)
+      {
+        std::tm* date = std::localtime(&old_wallet_ts);
```

Comment:
localtime might be problematic. If the "dated" seed word is generated from UTC time, it should restore in UTC time. Blocks in the blockchain are also stored in terms of UTC so that is preferable.